05.wc/wc.rb

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require 'optparse'
+
+PRINTING_WIDTH = 8
+
+def main
+  options = ARGV.getopts('lwc')
+  ARGV.none? ? print_pipelined_input(options) : print_specified_files(ARGV, options)
+end
+
+def print_pipelined_input(options)
+  stdin_lines = $stdin.readlines
+
+  stdin_prop = format_file_prop(stdin_lines)
+  stdin_prop[:file_name] = ''
+
+  print_file_props([stdin_prop], options)
+end
+
+def print_specified_files(files, options)
+  total_file_prop_count = { line_count: 0, word_count: 0, bytesize: 0, file_name: 'total' }
+
+  file_props =
+    files.map do |file|
+      file_lines = File.open(file, &:readlines)
+      file_prop = format_file_prop(file_lines, total_file_prop_count)
```

Comment:
`total_file_prop_count`は第2の戻り値を格納するような役割になってますね。
引数で渡したデータを破壊的に変更するのはあまりお行儀が良くないです。
[\!が付く破壊的メソッドの使用は原則として避ける \| FBC](https://bootcamp.fjord.jp/pages/do-not-use-destructive-method)

```suggestion
      file_prop = format_file_prop(file_lines)
      total_file_prop_count[:line_count] += file_prop[:line_count]
```

みたいな感じで、totalの加算はこのメソッド内で実施した方がベターです。